src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1251,4 +1251,691 @@ static class C
+
+        var expectedDiagnostics = new[]
+        {
+            // (8,36): warning CS9501: Argument 2 should not be passed with the 'ref' keyword
+            //         new C(default(object), ref i);
+            Diagnostic(ErrorCode.WRN_BadArgRef, "i").WithArguments("2", "ref").WithLocation(8, 36)
+        };
+
+        CompileAndVerify(source, expectedOutput: "object5", parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        CompileAndVerify(source, expectedOutput: "object5").VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Fact]
+    public void RefReadonlyParameter_RefArgument_OverloadResolution_03()
+    {
+        var source = """
+            class C
+            {
+                static string M1(object o, in int i) => "object" + i;
+                static string M1(C c, ref readonly int i) => "c" + i;
+                static void Main()
+                {
+                    int i = 5;
+                    System.Console.WriteLine(M1(null, ref i));
+                }
+            }
+            """;
+        CompileAndVerify(source, expectedOutput: "c5").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void RefReadonlyParameter_RefArgument_OverloadResolution_03_Ctor()
+    {
+        var source = """
+            class C
+            {
+                private C(object o, in int i) => System.Console.WriteLine("object" + i);
+                private C(C c, ref readonly int i) => System.Console.WriteLine("c" + i);
+                static void Main()
+                {
+                    int i = 5;
+                    new C(null, ref i);
+                }
+            }
+            """;
+        CompileAndVerify(source, expectedOutput: "c5").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void RefReadonlyParameter_PlainArgument_OverloadResolution()
+    {
+        var source = """
+            class C
+            {
+                static string M1(ref readonly int i) => "ref readonly" + i;
+                static string M1(int i) => "plain" + i;
+                static void Main()
+                {
+                    int i = 5;
+                    System.Console.Write(M1(i));
+                    System.Console.Write(M1(6));
+                }
+            }
+            """;
+        CompileAndVerify(source, expectedOutput: "plain5plain6").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void Invocation_VirtualMethod()
+    {
+        var source = """
+            class C
+            {
+                protected virtual void M(ref readonly int p) => System.Console.WriteLine(p);
+                static void Main()
+                {
+                    int x = 111;
+                    new C().M(ref x);
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, expectedOutput: "111");
+        verifier.VerifyDiagnostics();
+        verifier.VerifyMethodBody("C.Main", """
```

Comment:
>VerifyMethodBody [](http://example.com/codeflow?start=17&length=16)

Consider using VerifyIL helper instead, we are not really interested in sequence points. The shorter baselines. the better.

Reply:
I like that the IL is more readable with sequence points, I think that was also the main motivation behind adding `VerifyMethodBody` in [https://github.com/dotnet/roslyn/pull/66536](https://github.com/dotnet/roslyn/pull/66536). `VerifyIL` is even marked as obsolete. But these methods are short, so you're right, I will switch to shorter baselines.